packages/@react-stately/steplist/src/useStepListState.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaStepListProps, StepListState} from '@react-types/steplist';
+import {Collection, Node} from '@react-types/shared';
+import {Key, useMemo} from 'react';
+import {useControlledState} from '@react-stately/utils';
+import {useSingleSelectListState} from '@react-stately/list';
+
+export function useStepListState<T extends object>(props: AriaStepListProps<T>): StepListState<T> {
+  let selectState = useSingleSelectListState<T>(props);
+  let [lastCompletedStep, setLastCompletedStep] = useControlledState<Key>(props.lastCompletedStep, props.defaultCompletedStep, props.onLastCompletedStepChange);
```

Comment:
usually default matches up with the value prop, @devongovett was there a typo and this should be
lastCompletedStep and defaultLastCompletedStep?

Reply:
the default arg should probably be something along the lines of
props.defaultCompletedStep != null ? props.defaultCompletedStep : selectState.collection.getFirstKey()

though might need a bit more work if there are disabledKeys

Reply:
Hmmm, I think I disagree here, completed is different than selected.  A step becomes `completed` when one progresses beyond it.   If I don't pass `defaultCompletedStep`, I'm saying "The user is on the first step, and hasn't completed it yet". 

If we changed the code to default to the first key being marked as completed, then it would be saying "The user has completed the first step and is free to go to either the first or second step".

Reply:
Ah, yep, i misread that, thank you for explaining

Reply:
I corrected the name of the prop to `defaultLastCompletedStep` and fixed up a bunch of the code formatting/style issues in `useStepListState`